### ISSUE TITLE
Allow node options like --max_old_space_size

### ISSUE THREAD

**Issue by littleskunk** [Storj/storjshare-daemon#113]
### Package Versions

Replace the values below using the output from `storjshare --version`.

```
  * daemon: 2.5.1, core: 6.4.1, protocol: 1.1.0
```

Replace the values below using the output from `node --version`.

```
6.10.2
```

### Expected Behavior

Please describe the program's expected behavior.

By default node will use up to 1.5GB memory. On some systems it would be usefull to increase or decrease the limit.

For example:
`storjshare start --config '...' --node_args '--max_old_space_size=2048'` (see pm2 `--node_args`)

GUI user would need the same as environment variable or on the config file. For example like this `STORJ_NODE_ARGS=--max_old_space_size=2048`

### Actual Behavior

Please describe the program's actual behavior. Please include any stack traces
or log output in the back ticks below.

If I run `node --trace-gc storjshare-start.js --detached --config '...'` it will spawn a new node instance without my options. My node option will be ignored.

Note: I did not check restart, load or any other command. I would expect the same behavior for these commands.

**Comment by MeijeSibbel** [comment]
### Expected Behavior

Please describe the program's expected behavior. Include an example of your
usage code in the back ticks below if applicable.

```
Allow farmers to set MAX RAM usage at startup or as a configuration parameter
```

### Actual Behavior

Please describe the program's actual behavior. Please include any stack traces
or log output in the back ticks below.

```
Ok so one might ask why one would want this, let me explain. 

We have been stress testing the network over the last few days which causes a increase of traffic to the nodes and a subsequent increase in RAM usage per node. Most people don't take into account that a Storj Share node can consume +-4GB or more during peak times. What we are starting to see more and more during stress tests is that farmers will just turn their nodes OFF because the excessive RAM usage is causing their system to either become slow and unresponsive or just crash. This is a problem because we need farmers to be reliable, more so during times of abundant demand for the Storage. As demand will increase farmers with for example 4x cores and 4GB of RAM will run out of RAM almost immediately. 

Now there are a few ways to solve this in my opinion.

1. When adding a node, Storj checks how much RAM it needs (and how much is already used by other nodes) and if that amount of RAM is available. If it is available (minus 20-30% for the OS) it will allow the user to add another node.

2. Farmers can set the max RAM usage they want the node to consume at node setup (or a parameter in the config for the CLI).

3. Reduce the RAM usage to always fall 30-40% of total capacity (and make it adjustable), this would probably mean ignoring bigger shards and download/upload less shards at a time.  
```

**Comment by MeijeSibbel** [comment]
as an example:

![capture](https://user-images.githubusercontent.com/11735738/29243088-f3f0f68c-7f8e-11e7-82cf-3cd6ec5551d2.PNG)

**Comment by littleskunk** [comment]
That example is another issue. The memory usage should never go that high.

**Comment by egorpavlikhin** [comment]
Since you asked I'm posting it here too

![image](https://user-images.githubusercontent.com/1160707/29243131-a1533d24-7fa0-11e7-89ac-582b04052af0.png)

**Comment by littleskunk** [comment]
That belongs to https://github.com/Storj/storjshare-daemon/issues/114

**Comment by braydonf** [comment]
You can run without process management via `./bin/storjshare-start -d -c /some/path/config/storj-share/farmer-16/config.json` which should handle arbitrary nodejs arguments.

**Comment by littleskunk** [comment]
Even -d (detached) will start a new node process and ignore my options.

https://github.com/Storj/storjshare-daemon/blob/40eefe160966e52233e575cd2c6cc58749c10e8f/bin/storjshare-start.js#L40

**Comment by littleskunk** [comment]
Solved by nodejs v8 soon available as LTS version: https://medium.com/the-node-js-collection/node-options-has-landed-in-8-x-5fba57af703d
